src/guiguts/tools/pptxt.py

```diff
@@ -0,0 +1,6738 @@
+            new_buffer = True
+
+        if len(line) == 0:
+            # End of paragraph
+
+            if len(buffer) > 0:
+                paras.append(buffer)
+                buffer = ""
+                new_buffer = True
+
+                # Get next line from book
+
+                continue
+
+        else:
+            if not new_buffer:
+                # As buffer is not empty need to prefix lines with a separator
+
+                line = " " + line
+
+            else:
+                # Buffer should be empty. First line being appended needs no separator
+
+                pass
+
+        # Append line to buffer
+        buffer = buffer + line
+        new_buffer = False
+
+    # End of book. If anything in the buffer add it to 'paras' string
+
+    if len(buffer) > 0:
+        paras.append(buffer)
+
+    ###################################################
+    # We're done reading the input. Start processing it.
+    ###################################################
+
+    # Prefix output with a divider line. Done here as each
+    # check below only writes a divider when it's finished.
+
+    print(" ", file=string_out)
+    template = "{}"
+    print(template.format("-" * 80), file=string_out)
+
+    # The checks are run in the following order...
+
+    html_check()
+    uniocode_numeric_character_check()
+    asterisk_check()
+    adjacent_spaces()
+    trailing_spaces()
+    weird_characters()
+    spacing_check()
+    short_lines_check()
+    long_lines_check()
+    repeated_words_check()
+    ellipses_check()
+    dash_review()
+    curly_quote_check()
+    scanno_check()
+    # This final one does multiple checks.
+    specials_check()
+
+
+#########################################
+#
+# MAIN PROGRAM
+#
+#########################################
+if __name__ == "__main__":
+    if platform.system() == "Windows":
+        os.system("color")
+
+    # Usage and command line arguments
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "INPUT", type=lambda x: is_valid_file(parser, x), help="input text file"
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        help="report on each check carried out",
+        type=argparse.FileType("w", encoding="UTF-8"),
+    )
+    parser.add_argument(
+        "-d",
+        "--decorate",
+        help="use colour to highlight in report",
+        action="store_true",
+    )
+    parser.add_argument("-v", "--verbose", help="verbose output", action="store_true")
+    args = parser.parse_args()
+
+    # Positional argument - input text
+    srcfile = args.INPUT
+
+    # Default log output
+    log = sys.stdout
+
+    # Redirect log output if optional output file specified
+
+    if args.output:
+        logfile = args.output.name
+        log = open(logfile, "w", encoding="utf-8")
+
+    log.reconfigure(encoding="utf-8")  # type: ignore[attr-defined]
+
+    # Decorate (aka 'highlighting') is an optional argument
+
+    highlight = False
```

Comment:
It's common to use the `arg.` attributes directly rather than post-process them like this. so:
```python
pptxt(string_in, string_out, verbose=args.verbose, highlight=args.decorate)
```